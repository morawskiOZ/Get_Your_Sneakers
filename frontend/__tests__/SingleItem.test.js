import CartCount from "../components/CartCount"
import { shallow, mount } from "enzyme"
import toJSON from "enzyme-to-json"
import SingleItem, { SINGLE_ITEM_QUERY } from "../components/SingleItem"
import wait from "waait"
import { MockedProvider } from "react-apollo/test-utils"
import { fakeItem } from "../lib/testUtils"

describe("<SingleItem />", () => {
  it("redners with proper data", async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        result: {
          data: {
            item: fakeItem()
          }
        }
      }
    ]
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    )
    expect(wrapper.text()).toContain("Loading...")
    await wait()
    wrapper.update()
    expect(toJSON(wrapper.find("h2"))).toMatchSnapshot()
  })
  it("errors with a not found item", async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        result: {
          errors: [{ message: "Item not found" }]
        }
      }
    ]
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    )
    await wait()
    wrapper.update()
    const item = wrapper.find('[data-test="graphql-error"]')
    expect(item.text()).toContain("Shoot!Item not found")
    expect(toJSON(item)).toMatchSnapshot()
  })
})
