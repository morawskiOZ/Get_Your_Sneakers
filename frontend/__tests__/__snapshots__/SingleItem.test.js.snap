// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> errors with a not found item 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Shoot!
  </strong>
  Item not found
</p>
`;

exports[`<SingleItem /> redners with proper data 1`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;
